spec/Classes.tex

```diff
@@ -739,37 +1063,348 @@ \subsection{The Compiler-Generated Constructor}
+    super.init(x, y); // compiler-generated initializer
+    foo(); // Parent.foo()
+    this.z = z;
+    this.complete();
+    foo(); // Child.foo()
+  }
+
+  override proc foo() {
+    writeln("Child.foo");
+  }
+}
+
+var c = new Child(1.0, 2.0, 3.0);
+\end{chapel}
+\begin{chapelpost}
+writeln(c);
+\end{chapelpost}
+\begin{chapelcompopts}
+--no-warnings
+\end{chapelcompopts}
+Once the parent's initializer is finished, the parent method
+\chpl{foo} may be called. After the \chpl{complete} method is invoked, a call to \chpl{foo}
+resolves to the Child's overriden implementation:
+\begin{chapelprintoutput}{}
+Parent.foo
+Child.foo
+{x = 1.0, y = 2.0, z = 3.0}
+\end{chapelprintoutput}
+\end{chapelexample}
+
+\subsubsection{The Compiler Generated Initializer for Inheriting Classes}
+\label{The_Compiler_Generated_Initializer_for_Inheriting_Classes}
+\index{classes!initializers!initializer-inheritance!compiler-generated}
+
+The compiler-generated initializer for inheriting classes will have arguments
+with default values and names based on the field declarations in the parent
+class. Formals for the parent type will be listed before formals for the
+child type.
+
+\begin{chapelexample}{compilerGeneratedInheritanceInit.chpl}
+In the following code:
+\begin{chapel}
+class Parent {
+  var x, y: real;
+}
+
+class Child : Parent {
+  var z : real;
+}
+
+var A = new Child();
+var B = new Child(1.0, 2.0, 3.0);
+var C = new Child(y=10.0);
+\end{chapel}
+\begin{chapelpost}
+writeln(A);
+writeln(B);
+writeln(C);
+\end{chapelpost}
+\begin{chapelcompopts}
+--no-warnings
+\end{chapelcompopts}
+\begin{chapeloutput}
+{x = 0.0, y = 0.0, z = 0.0}
+{x = 1.0, y = 2.0, z = 3.0}
+{x = 0.0, y = 10.0, z = 0.0}
+\end{chapeloutput}
+Any \chpl{new} expressions using the \chpl{Child} type can invoke an
+initializer with three formals named \chpl{x}, \chpl{y}, and \chpl{z} that all
+have default values based on their types.
+\end{chapelexample}
+
+\subsubsection{The postinit Method for Inheriting Classes}
+\label{The_postinit_Method_for_Inheriting_Classes}
+\index{classes!initializers!initializer-inheritance!postinit}
+
+The \chpl{postinit} method on inheriting classes allows users to invoke
+the parent's \chpl{postinit} method using the \chpl{super} keyword. If the
+user does not explicitly invoke the parent's \chpl{postinit}, the compiler
+will insert the call at the top of the user's \chpl{postinit} method. If the
+parent type has a \chpl{postinit} method but the inheriting class does not,
+the compiler will generate a \chpl{postinit} method that simply invokes the
+parent's \chpl{postinit} method.
+
+\begin{chapelexample}{inheritancePostinit.chpl}
+In the following code:
+\begin{chapel}
+class Parent {
+  var a, b : real;
+  proc postinit() {
+    writeln("Parent.postinit: ", a, ", ", b);
+  }
+}
+
+class Child : Parent {
+  var x, y : real;
+  proc postinit() {
+    // compiler inserts "super.postinit();"
+    writeln("Child.postinit: ", a, ", ", b);
```

Comment:
Did you mean to have this one refer to `x` and `y` instead of the parent's fields?

Reply:
That would also remove the need for the writeln at the end of the example